lib/chef/provider/aws_cloudsearch_domain.rb

```diff
@@ -0,0 +1,160 @@
+require 'chef/provisioning/aws_driver/aws_provider'
+
+class Chef::Provider::AwsCloudsearchDomain < Chef::Provisioning::AWSDriver::AWSProvider
+
+  def create_aws_object
+    domain = nil # define here to ensure it is available outside of the coverge_by scope
+    converge_by "Creating new CloudSearch domain #{new_resource.domain_name}" do
+      domain = create_domain
+    end
+
+    update_aws_object(domain)
+  end
+
+  def destroy_aws_object(instance)
```

Comment:
The word `instance` actually refers to an AWS [instance](http://docs.aws.amazon.com/AWSRubySDK/latest/AWS/EC2/Instance.html).  I think you should replace `instance` with `domain`